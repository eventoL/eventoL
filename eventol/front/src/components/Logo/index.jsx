import React from 'react';
import PropTypes from 'prop-types';

import {INDEX_URL} from '../../utils/urls';
import {LOGO_HEADER_DEFAULT} from '../../utils/constants';

import './index.scss';


const Logo = ({logoHeader}) => (
  <div className='logo-component' id='logo'>
    <a href={INDEX_URL}>
      <img alt='header logo' src={logoHeader} />
    </a>
  </div>
);

Logo.propTypes = {
  logoHeader: PropTypes.string,
};

Logo.defaultProps = {
  logoHeader: LOGO_HEADER_DEFAULT,
};

export default Logo;
