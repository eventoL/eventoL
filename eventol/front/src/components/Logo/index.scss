@import '../../styles/index';

.logo-component {
  -webkit-transition: margin .125s ease;
  height: 41px;
  margin-right: 10px;
  transition: margin .125s ease;
  width: 151px;

  a {
    text-decoration: none;
  }

  svg {
    -webkit-transform: scale(.5);
    -webkit-transform-origin: 0 0;
    transform: scale(.5);
    transform-origin: 0 0;

    path {
      -webkit-transition: fill .125s ease;
      fill: $eventol-color;
      transition: fill .125s ease;
    }
  }
}

.logo-component:hover {
  cursor: pointer;
  margin-top: 2px;

  svg {
    path {
      // sass-lint:disable no-important
      fill: $black !important;
      // sass-lint:enable no-important
    }
  }
}
