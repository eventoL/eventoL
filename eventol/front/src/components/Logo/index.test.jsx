import React from 'react';
import renderer from 'react-test-renderer';

import Logo from '.';

describe('Logo', () => {
  let component;
  let logoHeader;
  let tree;

  const getComponent = () => {
    component = renderer.create(<Logo logoHeader={logoHeader} />);
    return component;
  };

  beforeEach(() => {
    logoHeader = undefined;
    component = getComponent();
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('With custom logo', () => {
    test('Snapshot', () => {
      logoHeader = 'http://logoHeader/';
      component = getComponent();
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
