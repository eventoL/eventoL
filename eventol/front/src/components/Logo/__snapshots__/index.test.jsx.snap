// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Default Snapshot 1`] = `
<div
  className="logo-component"
  id="logo"
>
  <a
    href="/"
  >
    <img
      alt="header logo"
      src="/static/manager/img/eventol-white.png"
    />
  </a>
</div>
`;

exports[`Logo With custom logo Snapshot 1`] = `
<div
  className="logo-component"
  id="logo"
>
  <a
    href="/"
  >
    <img
      alt="header logo"
      src="http://logoHeader/"
    />
  </a>
</div>
`;
